### DISCUSSION TITLE
How to read varchar using Rust?

### DISCUSSION THREAD

**Discussion by rain2307** [duckdb/duckdb#14471] [Q&A] [answered]
I want to generate Rust bindings using a C++ interface. I don't know how to parse the queried result (`VARCHAR`) into `duckdb_string'

**Comment by Tishj** [comment, marked as answer]
I suggest you look through the code + tests
<https://github.com/duckdb/duckdb/blob/52b43b166091c82b3f04bf8af15f0ace18207a64/test/api/capi/test_capi_data_chunk.cpp#L260>

That test should cover it :+1:
